source/dagger/gameplay/ping_pong/player_scores.cpp

```diff
@@ -46,10 +46,17 @@ void PlayerScoresSystem::Run()
             ballOnField++;
         }
     }
-
     if (ballOnField == 0)
-    {
+    {   
+        
         // TODO: set speed to be random in both directions
-        CreatePingPongBall(s_TileSize, ColorRGBA(1, 1, 1, 1), { rand()%10 + 4,rand()%10 + 4,0 },   { 0,rand()%(s_FieldHeight / 2),0 });
+        if (k) {
+            CreatePingPongBall(s_TileSize, ColorRGBA(1, 1, 1, 1), { rand() % 10 + 5, rand() % 10 + 4,0 }, { 0,rand() % (s_FieldHeight / 2),0 });
+            k = false;
+        }
+        else {
+            CreatePingPongBall(s_TileSize, ColorRGBA(1, 1, 1, 1), { rand() % 10 - 15, rand() % 10 + 4,0 }, { 0,rand() % (s_FieldHeight / 2),0 });
+            k = true;
```

Comment:
This would be better as a single `CreatePingPongBall`, with a variable called `offset` for example, that will be `5` or `-15` added to `rand()`. After that you can say `k = !k;` with whatever name you choose for `k` and lower repetition.